graf2d/gpadv7/v7/inc/ROOT/RColor.hxx

```diff
@@ -16,257 +16,61 @@
+   using RGB_t = std::array<int, 3>;
 
-   /// If this is an ordinal in a palette, resolve the
-   float GetPaletteOrdinal() const;
+   RColor(int r, int g, int b) : RColor() { SetRGB(r, g, b); }
 
-   friend bool operator==(const RColor &lhs, const RColor &rhs)
+   RColor(int r, int g, int b, double alfa) : RColor()
    {
-      if (lhs.fKind != rhs.fKind)
-         return false;
-      switch (lhs.fKind) {
-      case EKind::kPalettePos:
-         return lhs.fRedOrPalettePos == rhs.fRedOrPalettePos;
-      case EKind::kRGBA:
-         return lhs.fRedOrPalettePos == rhs.fRedOrPalettePos && lhs.fGreen == rhs.fGreen && lhs.fBlue == rhs.fBlue &&
-             lhs.fAlpha == rhs.fAlpha;
-      case EKind::kAuto:
-         return true; // is that what we need?
-      }
-      return false;
-   }
-
-   /// For RGBA or auto colors, get the red component (0..1).
-   float GetRed() const {
-      if (AssertNotPalettePos())
-         return fRedOrPalettePos;
-      return 0.;
-   }
-
-   /// For RGBA or auto colors, get the green component (0..1).
-   float GetGreen() const {
-      if (AssertNotPalettePos())
-         return fGreen;
-      return 0.;
-   }
-
-   /// For RGBA or auto colors, get the blue component (0..1).
-   float GetBlue() const {
-      if (AssertNotPalettePos())
-         return fBlue;
-      return 0.;
-   }
-
-   /// For RGBA or auto colors, get the alpha component (0..1).
-   float GetAlpha() const {
-      if (AssertNotPalettePos())
-         return fAlpha;
-      return 0.;
-   }
-
-   /// For RGBA or auto colors, set the red component.
-   void SetRed(float r) {
-      if (AssertNotPalettePos())
-         fRedOrPalettePos = r;
-   }
-
-   /// For RGBA or auto colors, set the green component.
-   void SetGreen(float g) {
-      if (AssertNotPalettePos())
-         fGreen = g;
-   }
-
-   /// For RGBA or auto colors, set the blue component.
-   void SetBlue(float b) {
-      if (AssertNotPalettePos())
-         fBlue = b;
+      SetRGB(r, g, b);
+      SetAlpha(alfa);
    }
 
-   /// For RGBA or auto colors, set the alpha component.
-   void SetAlpha(float a) {
-      if (AssertNotPalettePos())
-         fAlpha = a;
-   }
-
-   /// For RGBA or auto colors, set the alpha component.
-   void SetAlpha(Alpha a) {
-      if (AssertNotPalettePos())
-         fAlpha = (float)a;
-   }
+   RColor(const RGB_t &rgb) : RColor() { SetRGB(rgb[0], rgb[1], rgb[2]); }
 
-   /// Return the Hue, Light, Saturation (HLS) definition of this RColor
-   void GetHLS(float &hue, float &light, float &satur) {
-      hue = light = satur = 0.;
-      if (AssertNotPalettePos()) {
-         float rnorm, gnorm, bnorm, minval, maxval, msum, mdiff;
-         minval = maxval =0 ;
+   std::string GetRGB() const { return GetValue<std::string>("rgb"); }
+   RColor &SetRGB(const std::string &_rgb) { SetValue("rgb", _rgb); return *this; }
+   RColor &SetRGB(int r, int g, int b) { return SetRGB(std::to_string(r) + "," + std::to_string(g) + "," + std::to_string(b)); }
```

Comment:
Why not `#rrggbb` format?

Reply:
May be - it would be most compact representation. But I propose change RColor after merging PR.